### PR TITLE
Added headless rendering tutorial

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

A tutorial that describes how to use headless rendering.

## Test it

You should be able to run through the tutorial.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: tutorials/headless_rendering.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+\page headless_rendering Headless Rendering 
+  
+It is often desirable to run simulation on a remote computer, such as
+a computer managed by cloud provider, in order to paralellize work or access
+specific compute resources. Simulated sensors that require GPU access has
```
Comment (chapulina) [inline tutorials/headless_rendering.md:5]:
```suggestion
specific compute resources. Simulated sensors that require GPU access have
```

File: tutorials/headless_rendering.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+historically been difficult to use on a remote computer due to opengl's
```
Comment (chapulina) [inline tutorials/headless_rendering.md:6]:
```suggestion
historically been difficult to use on a remote computer due to OpenGL's
```

File: tutorials/headless_rendering.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+X server requirement. This issue can be resolved through installation and
+proper configuration of X, but the steps can be complex and error prone.
+
+An easier solution is through the use of [EGL](https://www.khronos.org/egl), which allows for the the creation of rendering surfaces without an X server. Ignition Gazebo has added support for EGL via the `--headless-rendering` command line option. For example:
```
Comment (chapulina) [inline tutorials/headless_rendering.md:10]: We should mention that this only works with Ogre2

File: tutorials/headless_rendering.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+## AWS Example
+
+This example will guide you through the process of launching and
+configuring an AWS GPU instance with Gazebo running headless.
+
+1. Go to the [AWS EC2 service](https://console.aws.amazon.com/ec2)
+2. Click the `Launch Instance` button in the upper right.
+3. Select `Ubuntu Server` version 20.04 or greater from the AMI list.
+4. Choose a GPU enabled instance type, such as `g3.4xlarge`.
```
Comment (chapulina) [inline tutorials/headless_rendering.md:28]: Does it have to be a GPU instance? I understood one of the benefits of EGL was running on non-GPU instances without NVIDIA drivers.

Comment (nkoenig) [reply tutorials/headless_rendering.md:28]: @iche033 , does lack of a GPU cause EGL to use software rendering?

File: tutorials/headless_rendering.md (hunk 3 above)
Comment (nkoenig) [reply tutorials/headless_rendering.md]: I mentioned OGRE2 in a couple places, see [53597c7](https://github.com/ignitionrobotics/ign-gazebo/pull/1386/commits/53597c79590e39cc32392827edfad9e47814570e)

File: tutorials/headless_rendering.md (hunk 4 above)
Comment (iche033) [reply tutorials/headless_rendering.md:31]: yeah here's [more info](https://www.ogre3d.org/2021/02/06/ogre-2-2-5-cerberus-released-and-egl-headless-support) in ogre-next's post under Headless SW Rasterization. There is software rasterization through Mesa.

Comment (nkoenig) [reply tutorials/headless_rendering.md:36]: Added note about software rendering in [2f43048](https://github.com/ignitionrobotics/ign-gazebo/pull/1386/commits/2f4304881881d7c87103ee62420cf421be3ac9d4)

File: tutorials/headless_rendering.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+\page headless_rendering Headless Rendering 
+  
+It is often desirable to run simulation on a remote computer, such as
+a computer managed by cloud provider, in order to paralellize work or access
+specific compute resources. Simulated sensors that require GPU access have
+historically been difficult to use on a remote computer due to OpenGL's
+X server requirement. This issue can be resolved through installation and
```
[the remark this answers is not in the document]
Comment (nkoenig) [reply tutorials/headless_rendering.md]: Added in [d908152](https://github.com/ignitionrobotics/ign-gazebo/pull/1386/commits/d908152952e7363454a4bc406d0eb77d10bf64eb).

**Review by iche033 — APPROVED** [review]
looks good to me

**Review by chapulina — APPROVED** [review]
Just need to sign DCO on all commits, then merge

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**